src/Microsoft.Health.Dicom.SqlServer/Features/Store/SqlIndexDataStore.cs

```diff
@@ -5,284 +5,99 @@
-                                VLatest.DeletedInstance.Watermark);
-
-                            results.Add(new VersionedInstanceIdentifier(
-                                studyInstanceUid,
-                                seriesInstanceUid,
-                                sopInstanceUid,
-                                watermark));
-                        }
-                    }
-                    catch (SqlException ex)
-                    {
-                        throw new DataStoreException(ex);
-                    }
-                }
-            }
-
-            return results;
+            await InitInnerDataStoreAsync();
+            return await _innerDataStore.RetrieveDeletedInstancesAsync(batchSize, maxRetries, cancellationToken);
         }
 
-        public async Task DeleteDeletedInstanceAsync(VersionedInstanceIdentifier versionedInstanceIdentifier, CancellationToken cancellationToken = default)
+        public async Task UpdateInstanceIndexStatusAsync(VersionedInstanceIdentifier versionedInstanceIdentifier, IndexStatus status, CancellationToken cancellationToken)
         {
-            await _sqlServerIndexSchema.EnsureInitialized();
-
-            using (SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionFactoryWrapper.ObtainSqlConnectionWrapperAsync(cancellationToken, true))
-            using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand())
-            {
-                VLatest.DeleteDeletedInstance.PopulateCommand(
-                    sqlCommandWrapper,
-                    versionedInstanceIdentifier.StudyInstanceUid,
-                    versionedInstanceIdentifier.SeriesInstanceUid,
-                    versionedInstanceIdentifier.SopInstanceUid,
-                    versionedInstanceIdentifier.Version);
-
-                try
-                {
-                    await sqlCommandWrapper.ExecuteScalarAsync(cancellationToken);
-                }
-                catch (SqlException ex)
-                {
-                    throw new DataStoreException(ex);
-                }
-            }
+            await InitInnerDataStoreAsync();
+            await _innerDataStore.UpdateInstanceIndexStatusAsync(versionedInstanceIdentifier, status, cancellationToken);
         }
 
-        public async Task<int> IncrementDeletedInstanceRetryAsync(VersionedInstanceIdentifier versionedInstanceIdentifier, DateTimeOffset cleanupAfter, CancellationToken cancellationToken = default)
+        private async Task InitInnerDataStoreAsync()
         {
             await _sqlServerIndexSchema.EnsureInitialized();
 
-            using (SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionFactoryWrapper.ObtainSqlConnectionWrapperAsync(cancellationToken, true))
-            using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand())
+            if (_schemaInformation.Current == 1)
             {
```

Comment:
Instead of if else conditions, can this be a static mapping?

Reply:
sure, will change

Reply:
+1, you can keep a dictionary of version and class type mapping and use that